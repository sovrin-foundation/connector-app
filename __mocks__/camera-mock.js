// @flow
import React from 'react'
import { View } from 'react-native'

const constants = {
  Aspect: {},
  BarCodeType: {},
  Type: {},
  CaptureMode: {},
  CaptureTarget: {},
  CaptureQuality: {},
  Orientation: {},
  FlashMode: {},
  TorchMode: {},
}

// We don't need to type check external library
// and we don't need types for mock test stub
class MockCamera extends React.Component<any, void> {
  static constants = constants
  static checkVideoAuthorizationStatus = jest.fn(() => Promise.resolve(true))

  render() {
    return <View>{this.props.children}</View>
  }
}

MockCamera.constants = constants

export default MockCamera
