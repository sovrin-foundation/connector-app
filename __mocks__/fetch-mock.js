import { Headers, Response, Request } from 'node-fetch'

const fetch = jest.fn(() => {})
fetch.Headers = Headers
fetch.Request = Request
fetch.Response = Response

fetch.mockResponse = (body, init) =>
  fetch.mockImplementation(
    () =>
      new Promise((resolve, reject) =>
        process.nextTick(() => resolve(new Response(body, init)))
      )
  )

fetch.mockResponseOnce = (body, init) =>
  fetch.mockImplementationOnce(
    () =>
      new Promise((resolve, reject) =>
        process.nextTick(() => resolve(new Response(body, init)))
      )
  )

fetch.mockResponses = (...responses) =>
  responses.map(([body, init]) =>
    fetch.mockImplementationOnce(
      () =>
        new Promise((resolve, reject) =>
          process.nextTick(() => resolve(new Response(body, init)))
        )
    )
  )

export default fetch
