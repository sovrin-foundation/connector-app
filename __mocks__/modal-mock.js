// @flow
import React from 'react'

const MockModal = (props: any) =>
  React.createElement('Modal', props, props.children)

export default MockModal
