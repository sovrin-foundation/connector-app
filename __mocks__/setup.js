import React from 'react'
import { NativeModules } from 'react-native'
import fetch from './fetch-mock'
import mockCamera from './camera-mock'
import mockModal from './modal-mock'
import mockView from './view-mock'

// mock Math.random for <Loader />
const mockMath = Object.create(global.Math)
mockMath.random = () => 0.1
global.Math = mockMath

jest.mock('NativeAnimatedHelper')

// mock this module to allow react-navigation to mock Linking
jest.mock('Linking', () => ({
  addEventListener: jest.fn(),
  removeEventListener: jest.fn(),
  openURL: jest.fn(() => Promise.resolve()),
  canOpenURL: jest.fn(() => Promise.resolve()),
  getInitialURL: jest.fn(() => Promise.resolve()),
}))

jest.mock('react-navigation', () => ({
  createStackNavigator: jest.fn(() => mockView),
  createTabNavigator: jest.fn(() => mockView),
  TabBarTop: jest.fn(),
  isFocused: true,
  withNavigationFocus: jest.fn(() => mockView),
  NavigationActions: {
    reset() {
      return arguments
    },
    navigate() {
      return arguments
    },
  },
}))

jest.mock('react-native-firebase', () => ({
  notifications: jest.fn(() => ({
    onNotificationDisplayed: jest.fn(() => Promise.resolve()),
    onNotification: jest.fn(() => Promise.resolve()),
    onNotificationOpened: jest.fn(() => Promise.resolve()),
    getInitialNotification: jest.fn(() => Promise.resolve()),
  })),
  messaging: jest.fn(() => ({
    requestPermission: jest.fn(() => Promise.resolve()),
    onTokenRefresh: jest.fn(() => Promise.resolve()),
    getToken: jest.fn(() => Promise.resolve()),
  })),
}))

jest.mock('PixelRatio', () => ({
  get: jest.fn(() => 3), //PixelRatio.get() === 3 then iPhone 6 plus
  roundToNearestPixel: jest.fn(num => num),
}))

jest.mock('react-native-sensitive-info', () => {
  // todo: need to handle empty key-chain case
  const secureStorage = {
    ConnectMeKeyChain: {
      identifier: '26u37qFNUk3QwCUQ9ctest',
      phone: '6428900484',
      seed: 'S1StqtsAL4zkFZxmW57sL75zmufmVFQ3',
      pushComMethod: 'dehVox1KRqM:APA91bFWrJea1avml_ELw2MaH60abydtest',
    },
  }
  return {
    setItem: jest.fn((key, value, st) =>
      Promise.resolve((secureStorage['ConnectMeKeyChain'][key] = value))
    ),
    getItem: jest.fn(key =>
      Promise.resolve(secureStorage['ConnectMeKeyChain'][key])
    ),
    deleteItem: jest.fn(key =>
      Promise.resolve(delete secureStorage['ConnectMeKeyChain'][key])
    ),
  }
})

jest.mock('react-native-animatable', () => ({
  View: 'Animatable.View',
}))

jest.mock('react-native-camera', () => mockCamera)

jest.mock('Dimensions', () => ({
  get: jest.fn(() => ({
    width: 320,
    height: 540,
  })),
}))

jest.mock('react-native-touch-id', () => {
  return {
    authenticate: jest.fn(message => Promise.resolve()),
    isSupported: jest.fn(_ => Promise.resolve()),
  }
})

// jest.mock('react-native-sentry', () => ({
//   Sentry: {
//     install() {},
//     config() {
//       return { install() {} }
//     },
//     captureException(error) {},
//   },
//   SentryLog: {
//     Debug: 1,
//   },
// }))

jest.mock('AlertIOS', () => ({
  alert() {},
}))

global.fetch = fetch

jest.mock('Modal', () => mockModal)

jest.mock('react-native-modal', () => mockModal)

jest.mock('react-native-fetch-blob', () => ({
  fetch: jest.fn((type, url) => Promise.resolve()),
  fs: {
    dirs: {
      DocumentDir: '/var/application/DocumentDir',
    },
    exists: jest.fn(path => Promise.resolve(true)),
    unlink: jest.fn(path => Promise.resolve(true)),
    cp: jest.fn((source, destination) => Promise.resolve(destination)),
  },
}))

jest.mock('WebView', () => 'WebView')

jest.mock('InteractionManager', () => ({
  runAfterInteractions(fn) {
    fn()
  },
  createInteractionHandle: jest.fn(),
  clearInteractionHandle: jest.fn(),
}))

jest.mock('TextInput', () => 'TextInput')

jest.mock('moment', () =>
  jest.fn(date => ({
    format(format) {
      if (format === 'MM/DD/YYYY') {
        return '01/10/2018'
      }

      if (format === 'MMMM YYYY') {
        return 'January 2018'
      }

      return '2018-01-10T04:32:43+05:30'
    },
  }))
)

jest.mock('react-native-share', () => {})

jest.mock('react-native-branch', () => {
  return {
    ADD_TO_CART_EVENT: 'Add to Cart',
    ADD_TO_WISHLIST_EVENT: 'Add to Wishlist',
    PURCHASE_INITIATED_EVENT: 'Purchase Started',
    PURCHASED_EVENT: 'Purchased',
    REGISTER_VIEW_EVENT: 'View',
    SHARE_COMPLETED_EVENT: 'Share Completed',
    SHARE_INITIATED_EVENT: 'Share Started',

    STANDARD_EVENT_ADD_TO_CART: 'ADD_TO_CART',
    STANDARD_EVENT_ADD_TO_WISHLIST: 'ADD_TO_WISHLIST',
    STANDARD_EVENT_VIEW_CART: 'VIEW_CART',
    STANDARD_EVENT_INITIATE_PURCHASE: 'INITIATE_PURCHASE',
    STANDARD_EVENT_ADD_PAYMENT_INFO: 'ADD_PAYMENT_INFO',
    STANDARD_EVENT_PURCHASE: 'PURCHASE',
    STANDARD_EVENT_SPEND_CREDITS: 'SPEND_CREDITS',

    STANDARD_EVENT_SEARCH: 'SEARCH',
    STANDARD_EVENT_VIEW_ITEM: 'VIEW_ITEM',
    STANDARD_EVENT_VIEW_ITEMS: 'VIEW_ITEMS',
    STANDARD_EVENT_RATE: 'RATE',
    STANDARD_EVENT_SHARE: 'SHARE',

    STANDARD_EVENT_COMPLETE_REGISTRATION: 'COMPLETE_REGISTRATION',
    STANDARD_EVENT_COMPLETE_TUTORIAL: 'COMPLETE_TUTORIAL',
    STANDARD_EVENT_ACHIEVE_LEVEL: 'ACHIEVE_LEVEL',
    STANDARD_EVENT_UNLOCK_ACHIEVEMENT: 'UNLOCK_ACHIEVEMENT',

    // don't need any type checks for mocks, because they are not used
    // for type checking in code
    subscribe(cb: any) {},
  }
})

jest.mock('react-native-splash-screen', () => ({
  hide: jest.fn(),
}))

jest.mock('react-native-unique-id', () =>
  jest.fn(() => Promise.resolve('uniqueDeviceId'))
)

jest.mock('react-native-image-crop-picker', () => ({
  openPicker: jest.fn(() =>
    Promise.resolve({ path: '/var/application/DocumentDir/user-avatar.jpeg' })
  ),
}))

jest.mock('react-native-share', () => {})

// Add mock for RNIndy on NativeModules
// we don't want to mock NativeModules the way we are doing for all above
// because then we would have to mock everything else as well of NativeModules
// a lot of other modules depend on it, so we are adding RNIndy
// which is what we need to mock
NativeModules.RNIndy = {
  deserializeConnection: jest.fn(_ => Promise.resolve(1)),
  setWalletItem: jest.fn(_ => Promise.resolve(1)),
  deleteWalletItem: jest.fn(_ => Promise.resolve(1)),
  updateWalletItem: jest.fn(_ => Promise.resolve(1)),
  createWalletKey: jest.fn(_ => Promise.resolve('walletKey')),
  exitAppAndroid: jest.fn(),
}

jest.mock('react-native-version-number', () => ({
  appVersion: '1.0',
  buildVersion: '500',
}))

jest.mock('apptentive-react-native', () => ({
  Apptentive: {
    register: () => Promise.resolve(null),
    presentMessageCenter: jest.fn(),
    onAuthenticationFailed: reason => null,
  },
  ApptentiveConfiguration: () => null,
}))
jest.mock('BackHandler', () => ({
  addEventListener: jest.fn(),
  removeEventListener: jest.fn(),
}))
