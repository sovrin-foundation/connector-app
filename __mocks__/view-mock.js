// @flow
import React from 'react'
import { View } from 'react-native'

class MockView extends React.Component<any, void> {
  render() {
    return <View>{this.props.children}</View>
  }
}

export default MockView
