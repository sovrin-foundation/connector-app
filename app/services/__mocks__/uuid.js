// @flow
import { defaultUUID } from '../../../__mocks__/static-data'

export function uuid() {
  return defaultUUID
}
