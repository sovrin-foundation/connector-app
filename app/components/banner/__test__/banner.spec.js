// @flow
import React from 'react'
import 'react-native'
import renderer from 'react-test-renderer'
import { Provider } from 'react-redux'
import DefaultBanner, { Banner } from './../banner'
import moment from 'moment'
import { getStore, getNavigation } from './../../../../__mocks__/static-data'

describe('<Banner/>', () => {
  let props
  let store

  let getProps = () => ({
    walletBackup: jest.fn(),
    offline: false,
    showBanner: false,
    lastSuccessfulBackupTimeStamp: '',
    navigation: getNavigation(),
    promptBackupBanner: jest.fn(),
  })

  let Store = () => ({
    ...getStore(),
    getState: () => ({
      ...getStore().getState(),
      offline: {
        offline: true,
      },
    }),
  })

  let OfflineStore = () => ({
    ...getStore(),
    getState: () => ({
      ...getStore().getState(),
      offline: {
        offline: false,
      },
    }),
  })

  beforeEach(() => {
    props = getProps()
    store = Store()
  })

  it('should display nothing ', () => {
    const wrapper = renderer.create(<Banner {...props} />).toJSON()
    expect(wrapper).toMatchSnapshot()
  })

  it('should display offline banner', () => {
    const wrapper = renderer
      .create(
        <Provider store={store}>
          <DefaultBanner
            walletBackup={jest.fn()}
            offline={true}
            showBanner={false}
            lastSuccessfulBackupTimeStamp={''}
            navigation={getNavigation()}
            promptBackupBanner={jest.fn()}
          />
        </Provider>
      )
      .toJSON()
    expect(wrapper).toMatchSnapshot()
  })

  it('should display "Set up data recovery now!" FIRST_BACKUP_TITLE text', () => {
    store = {
      ...getStore(),
      getState: () => ({
        ...getStore().getState(),
        offline: {
          offline: false,
        },
        backup: {
          ...getStore().getState().backup,
          lastSuccessfulBackup: '',
        },
      }),
    }
    const wrapper = renderer
      .create(
        <Provider store={store}>
          <DefaultBanner {...props} offline={false} />
        </Provider>
      )
      .toJSON()
    expect(wrapper).toMatchSnapshot()
  })

  it('should display "You have new data that is not backed up" SUBSEQUENT_BACKUP_TITLE text', () => {
    store = OfflineStore()
    const wrapper = renderer
      .create(
        <Provider store={store}>
          <DefaultBanner {...props} offline={false} />
        </Provider>
      )
      .toJSON()
    expect(wrapper).toMatchSnapshot()
  })
})
