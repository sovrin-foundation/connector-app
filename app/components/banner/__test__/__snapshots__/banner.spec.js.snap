// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Banner/> should display "Set up data recovery now!" FIRST_BACKUP_TITLE text 1`] = `
<View
  style={Object {}}
>
  <View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel="wallet-backup-banner-label-touchable"
      accessibilityTraits={undefined}
      accessible={false}
      hitSlop={
        Object {
          "bottom": 20,
          "left": 20,
          "right": 20,
          "top": 20,
        }
      }
      horizontalSpace={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      row={true}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#d0021b",
          "flexDirection": "row",
          "paddingHorizontal": 15,
          "paddingVertical": 8,
        }
      }
      testID="wallet-backup-banner-label-touchable"
      vCenter={true}
      verticalSpace={true}
    >
      <View
        clearBg={true}
        iconStyle={
          Array [
            Object {
              "height": 25,
              "width": 29,
            },
            Object {
              "tintColor": "white",
            },
          ]
        }
        src={1}
        style={
          Object {
            "backgroundColor": "transparent",
          }
        }
      >
        <View
          style={Object {}}
        >
          <Image
            resizeMode="contain"
            source={1}
            style={
              Array [
                Object {
                  "height": 50,
                  "width": 50,
                },
                null,
                Object {
                  "height": 25,
                  "width": 29,
                },
                Object {
                  "tintColor": "white",
                },
              ]
            }
          />
        </View>
      </View>
      <View
        horizontalSpace={true}
        style={
          Object {
            "paddingHorizontal": 15,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontSize": 17,
              },
              Object {
                "fontFamily": "Lato",
              },
              Object {
                "backgroundColor": "#333333",
                "color": "#ffffff",
              },
              Object {
                "fontWeight": "bold",
              },
              null,
              Object {
                "backgroundColor": "transparent",
              },
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
            ]
          }
        >
          Set up data recovery now!
        </Text>
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontSize": 15,
              },
              Object {
                "fontFamily": "Lato",
              },
              Object {
                "backgroundColor": "#333333",
                "color": "#ffffff",
              },
              null,
              null,
              Object {
                "backgroundColor": "transparent",
              },
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
            ]
          }
        >
          2 easy steps, takes 3 minutes.
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`<Banner/> should display "You have new data that is not backed up" SUBSEQUENT_BACKUP_TITLE text 1`] = `
<View
  style={Object {}}
>
  <View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel="wallet-backup-banner-label-touchable"
      accessibilityTraits={undefined}
      accessible={false}
      hitSlop={
        Object {
          "bottom": 20,
          "left": 20,
          "right": 20,
          "top": 20,
        }
      }
      horizontalSpace={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      row={true}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#d0021b",
          "flexDirection": "row",
          "paddingHorizontal": 15,
          "paddingVertical": 8,
        }
      }
      testID="wallet-backup-banner-label-touchable"
      vCenter={true}
      verticalSpace={true}
    >
      <View
        clearBg={true}
        iconStyle={
          Array [
            Object {
              "height": 25,
              "width": 29,
            },
            Object {
              "tintColor": "white",
            },
          ]
        }
        src={1}
        style={
          Object {
            "backgroundColor": "transparent",
          }
        }
      >
        <View
          style={Object {}}
        >
          <Image
            resizeMode="contain"
            source={1}
            style={
              Array [
                Object {
                  "height": 50,
                  "width": 50,
                },
                null,
                Object {
                  "height": 25,
                  "width": 29,
                },
                Object {
                  "tintColor": "white",
                },
              ]
            }
          />
        </View>
      </View>
      <View
        horizontalSpace={true}
        style={
          Object {
            "paddingHorizontal": 15,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontSize": 17,
              },
              Object {
                "fontFamily": "Lato",
              },
              Object {
                "backgroundColor": "#333333",
                "color": "#ffffff",
              },
              Object {
                "fontWeight": "bold",
              },
              null,
              Object {
                "backgroundColor": "transparent",
              },
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
            ]
          }
        >
          You have new data that is not backed up.
        </Text>
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontSize": 15,
              },
              Object {
                "fontFamily": "Lato",
              },
              Object {
                "backgroundColor": "#333333",
                "color": "#ffffff",
              },
              null,
              null,
              Object {
                "backgroundColor": "transparent",
              },
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
            ]
          }
        >
          Create another backup so you don't lose anything.
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`<Banner/> should display nothing  1`] = `null`;

exports[`<Banner/> should display offline banner 1`] = `
<View
  style={Object {}}
>
  <View
    accessibilityLabel="wallet-backup-banner-label"
    accessible={true}
    horizontalSpace={true}
    row={true}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFD5DB",
        "flexDirection": "row",
        "paddingHorizontal": 15,
        "paddingVertical": 8,
      }
    }
    testID="wallet-backup-banner-label"
    vCenter={true}
    verticalSpace={true}
  >
    <View
      horizontalSpace={true}
      style={
        Object {
          "paddingHorizontal": 15,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontSize": 17,
            },
            Object {
              "fontFamily": "Lato",
            },
            Object {
              "backgroundColor": "#333333",
              "color": "#ffffff",
            },
            Object {
              "fontWeight": "bold",
            },
            Object {
              "textAlign": "center",
            },
            Object {
              "backgroundColor": "transparent",
            },
            null,
            Object {
              "color": "#d0021b",
            },
            null,
            null,
            null,
            null,
            null,
            null,
          ]
        }
      >
        You are not connected to the internet
      </Text>
    </View>
  </View>
</View>
`;
